### PR TITLE
Document stategic merge patch.

### PR DESCRIPTION
**Pull request by mikechengwei**
Fix: #3816 

**Release Note Template (will be copied)**

```release-note:REPLACEME
Add a document describe how the above "strategic merge patch" works.
```

### CODE REVIEW

File: Documentation/user-guides/strategic-merge-patch.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Strategic merge patch
+
+Some users find some unreasonable container configurations,they want to have modifications on it.
```
Comment (lilic) [inline Documentation/user-guides/strategic-merge-patch.md:3]:
```suggestion
When users need to apply a specific configuration to containers that we do not support or do not currently exist, merge patch can be used.
```

File: Documentation/user-guides/strategic-merge-patch.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+This document describes how to overwrite the container configuration generated by the operator by merging patches.
+
+## How the "strategic merge patch" works
+	
+Operator support `containers` field in `PrometheusSpec` and `AlertmanagerSpec` configuration.
+This field allows injecting additional containers,and the existing configuration can be overwritten by sharing the same conatiner name.
```
Comment (lilic) [inline Documentation/user-guides/strategic-merge-patch.md:9]:
```suggestion
This field allows injecting additional containers, and the existing configuration can be overwritten by sharing the same container name.
```

File: Documentation/user-guides/strategic-merge-patch.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+### Merge patch prometheus example
```
Comment (lilic) [inline Documentation/user-guides/strategic-merge-patch.md:11]:
```suggestion
### Merge patch Prometheus example
```

File: Documentation/user-guides/strategic-merge-patch.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+Overwrite prometheus container readinessProbe  `FailureThreshold` configuration.
```
Comment (lilic) [inline Documentation/user-guides/strategic-merge-patch.md:13]:
```suggestion
The following is an example on how to overwrite Prometheus container `readinessProbe`  `FailureThreshold` configuration.
```

**Review by lilic — COMMENTED** [review]
Thank you for the PR! Few suggestions!

cc @prometheus-operator/prometheus-operator-reviewers please have another look.

File: Documentation/user-guides/strategic-merge-patch.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
+# Strategic merge patch
+
+When users need to apply a specific configuration to containers that we do not support or do not currently exist, merge patch can be used.
+This document describes how to overwrite the container configuration generated by the operator by merging patches.
+
+## How the "strategic merge patch" works
+	
+Operator support `containers` field in `PrometheusSpec` and `AlertmanagerSpec` configuration.
```
Comment (simonpasquier) [inline Documentation/user-guides/strategic-merge-patch.md:8]:
```suggestion
The operator supports `containers` field in `PrometheusSpec`, `AlertmanagerSpec` and `ThanosRulerSpec` configuration.
```

File: Documentation/user-guides/strategic-merge-patch.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+This field allows injecting additional containers, and the existing configuration can be overwritten by sharing the same container name.
+
+### Merge patch Prometheus example
+
+The following is an example on how to overwrite Prometheus container `readinessProbe`  `FailureThreshold` configuration.
```
Comment (simonpasquier) [inline Documentation/user-guides/strategic-merge-patch.md:13]:
```suggestion
The following manifest overwrites the `failureThreshold` value of the readiness probe for the Prometheus container.
```

File: Documentation/user-guides/strategic-merge-patch.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  name: self
+  namespace: default
+spec:
+  containers:
+  - name: prometheus
+    readinessProbe
+      failureThreshold: 500
+```
+
+### Merge patch alertmanager example
```
Comment (simonpasquier) [inline Documentation/user-guides/strategic-merge-patch.md:30]:
```suggestion
### Merge patch example for Alertmanager
```

File: Documentation/user-guides/strategic-merge-patch.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+When users need to apply a specific configuration to containers that we do not support or do not currently exist, merge patch can be used.
+This document describes how to overwrite the container configuration generated by the operator by merging patches.
+
+## How the "strategic merge patch" works
+	
+Operator support `containers` field in `PrometheusSpec` and `AlertmanagerSpec` configuration.
+This field allows injecting additional containers, and the existing configuration can be overwritten by sharing the same container name.
+
+### Merge patch Prometheus example
```
Comment (simonpasquier) [inline Documentation/user-guides/strategic-merge-patch.md:11]:
```suggestion
### Merge patch example of Prometheus
```

File: Documentation/user-guides/strategic-merge-patch.md [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+kind: Prometheus
+metadata:
+  labels:
+    prometheus: self
+  name: self
+  namespace: default
+spec:
+  containers:
+  - name: prometheus
+    readinessProbe
```
Comment (simonpasquier) [inline Documentation/user-guides/strategic-merge-patch.md:26]:
```suggestion
    readinessProbe:
```

File: Documentation/user-guides/strategic-merge-patch.md [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+      failureThreshold: 500
+```
+
+### Merge patch alertmanager example
+
+Overwrite alertmanager container readinessProbe  `FailureThreshold` configuration.
```
Comment (simonpasquier) [inline Documentation/user-guides/strategic-merge-patch.md:32]:
```suggestion
The following manifest overwrites the `failureThreshold` values of the readiness and liveness probes for the Alertmanager container.
```

File: Documentation/user-guides/strategic-merge-patch.md [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+  labels:
+    alertmanager: main
+  name: main
+  namespace: monitoring
+spec:
+  containers:
+  - name: alertmanager
+    livenessProbe:
+      failureThreshold: 5
+    readinessProbe
```
Comment (simonpasquier) [inline Documentation/user-guides/strategic-merge-patch.md:47]:
```suggestion
    readinessProbe:
```

**Review by simonpasquier — COMMENTED** [review]
Only few nits, thanks for doing this!

**Review by simonpasquier — APPROVED** [review]
lgtm

**Comment by lilic** [comment]
Thank you! lgtm

**Pull request merged**